pds_doi_core/db/doi_database.py

```diff
@@ -243,6 +273,20 @@ def insert_row(self, db_name, table_name, dict_row, drop_exist_table_flag=False)
 
         return 1
 
+    def update_previous_records_is_latest_field(self,dict_row):
+        '''Update all records that share the same lid,vid before the current update_date field in dict_row
+           by setting the is_latest field to False (or 0).'''
+
+        logger.debug(f"dict_row {dict_row}")
+
+        # Create the query string with the criteria for lid/vid and update_date fields.
+        query_string = self.create_q_string_for_transaction_update_is_latest_field(self.m_default_table_name, dict_row)
+
+        self.m_my_conn.execute(query_string)
+        self.m_my_conn.commit()
```

Comment:
This part should be done in the same commit has the creation of the new row. In case something happen (database connexion failure) .

Reply:
The two actions of inserting a new row and updating previous rows have now been grouped together so the commit now applies to both actions.

Reply:
great